herd/AArch64Sem.ml

```diff
@@ -2101,10 +2093,40 @@ module Make
             (* CAS does not generate an Explicit Write Effect          *)
             (* It is IMPLEMENTATION SPECIFIC if there is an update to  *)
             (*                                the dirty bit of the TTD *)
-            (* Note: the combination of dir=Dir.R and updatedb=true    *)
-            (*                    triggers an alternative in check_ptw *)
-            lift_memop rn Dir.R true memtag mop_fail_no_wb (to_perms "rw" sz)
-              (read_reg_addr rn ii) (read_reg_data_sz sz rt ii) an ii
+            let proc = ii.AArch64.proc in
+            let tthm = dirty.DirtyBit.tthm proc
+            and hd = dirty.DirtyBit.hd proc in
+            let may_update_db = tthm && hd in
+            let ma = read_reg_addr rn ii
+            and action updatedb checked ma =
+              lift_memop rn Dir.R updatedb checked mop_fail_no_wb (to_perms "rw" sz)
+                ma (read_reg_data_sz sz rt ii) an ii in
+
+            if memtag && C.mte_store_only then
+              M.altT (
+                (* If FEAT_MTE_STORE_ONLY is implemented it is              *)
+                (* CONSTRAINED UNPREDICTABLE whether the Tag Check          *)
+                (* operation is performed.                                  *)
+                (*                                                          *)
```

Comment:
I am nitpicking but this comment applies for both parts of altT so you might want to move it above the call to altT?